app/src/userpages/modules/transactionHistory/actions.js

```diff
@@ -90,12 +90,9 @@ export const showEvents = () => (dispatch: Function, getState: () => StoreState)
                     entities.contractProducts &&
                     entities.contractProducts[transaction.productId]
                 ))
-                .reduce(
-                    (result, transaction: TransactionEntity) =>
-                        // $FlowFixMe
-                        (result.includes(transaction.productId) ? result : [...result, transaction.productId]),
-                    [],
-                )
+                .reduce((result, transaction: TransactionEntity) => (
+                    result.includes(transaction.productId) ? result : [...result, (transaction.productId || '')]
+                ), [])
```

Comment:
Not important really but this `reduce` should probably go away in favour of using  a`Set` to do the deduping:

```js
Array.from(new Set(productsToFetch.map(({ productId }) => productId))
```